runtime/jcl/common/sun_misc_Unsafe.cpp

```diff
@@ -96,19 +95,72 @@ Java_sun_misc_Unsafe_defineAnonymousClass(JNIEnv *env, jobject receiver, jclass
 		hostClassLoader = vm->systemClassLoader->classLoaderObject;
 	}
 	jobject hostClassLoaderLocalRef = vmFuncs->j9jni_createLocalRef(env, hostClassLoader);
+
+	J9ClassPatchMap cpPatchMap = {0, NULL};
+	j9array_t patchArray = NULL;
+	PORT_ACCESS_FROM_ENV(env);
+
+	U_16 indexMap[BUFFER_SIZE];
+	if (constPatches != NULL) {
+		patchArray = (j9array_t)J9_JNI_UNWRAP_REFERENCE(constPatches);
+		cpPatchMap.size = (U_16)J9INDEXABLEOBJECT_SIZE(currentThread, patchArray);
+		if (cpPatchMap.size <= BUFFER_SIZE) {
+			cpPatchMap.indexMap = indexMap;
+		} else {
+			cpPatchMap.indexMap = (U_16 *)j9mem_allocate_memory(cpPatchMap.size * sizeof(U_16), J9MEM_CATEGORY_VM);
+
+			if (cpPatchMap.indexMap == NULL) {
+				vmFuncs->throwNativeOOMError(env, 0, 0);
+				vmFuncs->internalExitVMToJNI(currentThread);
+				return NULL;
+			}
+		}
+	}
+
 	vmFuncs->internalExitVMToJNI(currentThread);
 
 	jsize length = env->GetArrayLength(bytecodes);
 
 	/* acquires access internally */
-	jclass anonClass = defineClassCommon(env, hostClassLoaderLocalRef, NULL,bytecodes, 0, length, protectionDomainLocalRef, J9_FINDCLASS_FLAG_UNSAFE | J9_FINDCLASS_FLAG_ANON, hostClazz);
+	jclass anonClass = defineClassCommon(env, hostClassLoaderLocalRef, NULL,bytecodes, 0, length, protectionDomainLocalRef, J9_FINDCLASS_FLAG_UNSAFE | J9_FINDCLASS_FLAG_ANON, hostClazz, &cpPatchMap);
 	if (env->ExceptionCheck()) {
 		return NULL;
 	} else if (NULL == anonClass) {
 		throwNewInternalError(env, NULL);
 		return NULL;
 	}
 
+	if (constPatches != NULL) {
+		vmFuncs->internalEnterVMFromJNI(currentThread);
+		J9Class *clazz = J9VM_J9CLASS_FROM_HEAPCLASS(currentThread, J9_JNI_UNWRAP_REFERENCE(anonClass));
+		U_32 *cpShapeDescription = J9ROMCLASS_CPSHAPEDESCRIPTION(clazz->romClass);
+		J9ConstantPool *ramCP = J9_CP_FROM_CLASS(clazz);
+		J9ROMConstantPoolItem *romCP = ramCP->romConstantPool;
+
+		/* Get J9 constantpool mapped item for patch item, only support patching STRING entries has been added */
+		for (U_16 i = 0; i < cpPatchMap.size; i++) {
+			j9object_t item = J9JAVAARRAYOFOBJECT_LOAD(currentThread, patchArray, i);
+			if ((item != NULL) && (J9_CP_TYPE(cpShapeDescription, cpPatchMap.indexMap[i]) == J9CPTYPE_STRING)) {
+				J9UTF8 *romString = J9ROMSTRINGREF_UTF8DATA((J9ROMStringRef *)&romCP[cpPatchMap.indexMap[i]]);
+
+				/* For each patch object, search the RAM constantpool for identical string entries */
+				for (U_16 j = 1; j < clazz->romClass->ramConstantPoolCount; j++) {
+					if ((J9_CP_TYPE(cpShapeDescription, j) == J9CPTYPE_STRING)
+						&& J9UTF8_EQUALS(romString, J9ROMSTRINGREF_UTF8DATA((J9ROMStringRef *)&romCP[j]))
+					) {
+						J9RAMStringRef *ramStringRef = ((J9RAMStringRef *)ramCP) + j;
+						J9STATIC_OBJECT_STORE(currentThread, clazz, &ramStringRef->stringObject, item);
+					}
+				}
+			}
+		}
+
+		if (cpPatchMap.size > BUFFER_SIZE) {
+			j9mem_free_memory(cpPatchMap.indexMap);
```

Comment:
It's a good practice to null out fields that have been released even when the end of the function is so close.  Future changes to the file may accidentally reuse the freed mem if we don't null it now.